contracts/VaultV3.vy

```diff
@@ -187,73 +200,80 @@ def withdraw(_shares: uint256, _recipient: address, _strategies: DynArray[addres
     return self._amountForShares(10 ** self.decimals)
 
+
 @external
 def sharesForAmount(amount: uint256) -> uint256:
     return self._sharesForAmount(amount)
 
+
 @external
 def amountForShares(shares: uint256) -> uint256:
     return self._amountForShares(shares)
 
+
 # STRATEGY MANAGEMENT FUNCTIONS #
 @external
 def addStrategy(new_strategy: address):
-   # TODO: permissioned: STRATEGY_MANAGER
-   assert new_strategy != ZERO_ADDRESS
-   assert self == IStrategy(new_strategy).vault()
-   assert self.strategies[new_strategy].activation == 0
-   assert IStrategy(new_strategy).asset() != self.asset.address
-   
-   self.strategies[new_strategy] = StrategyParams({
-      activation: block.timestamp,
-      currentDebt: 0,
-      maxDebt: 0
-   })
-
-   log StrategyAdded(new_strategy)
-
-   return
+    # TODO: permissioned: STRATEGY_MANAGER
+    assert new_strategy != ZERO_ADDRESS, "strategy cannot be zero address"
+    assert IStrategy(new_strategy).asset() == self.asset.address, "invalid asset"
+    assert IStrategy(new_strategy).vault() == self, "invalid vault"
+    assert self.strategies[new_strategy].activation == 0, "strategy already active"
+
+    self.strategies[new_strategy] = StrategyParams({
+        activation: block.timestamp,
+        currentDebt: 0,
+        maxDebt: 0
+    })
+
+    log StrategyAdded(new_strategy)
+
 
 @internal
 def _revokeStrategy(old_strategy: address):
-   # TODO: permissioned: STRATEGY_MANAGER
-   assert self.strategies[old_strategy].activation != 0
-   # NOTE: strategy needs to have 0 debt to be revoked
-   assert self.strategies[old_strategy].currentDebt == 0
-
-   # NOTE: strategy params are set to 0 (warning: it can be readded)
-   self.strategies[old_strategy] = StrategyParams({
-       activation: 0,
-       currentDebt: 0,
-       maxDebt: 0
-   })
+    # TODO: permissioned: STRATEGY_MANAGER
+    assert self.strategies[old_strategy].activation != 0, "strategy not active"
+    # NOTE: strategy needs to have 0 debt to be revoked
+    assert self.strategies[old_strategy].currentDebt == 0, "strategy has debt"
+
+    # NOTE: strategy params are set to 0 (warning: it can be readded)
+    self.strategies[old_strategy] = StrategyParams({
+        activation: 0,
+        currentDebt: 0,
+        maxDebt: 0
+    })
 
-   log StrategyRevoked(old_strategy)
+    log StrategyRevoked(old_strategy)
 
-   return
 
 @external
 def revokeStrategy(old_strategy: address):
     self._revokeStrategy(old_strategy)
 
+
 @external
 def migrateStrategy(new_strategy: address, old_strategy: address):
-   # TODO: permissioned: STRATEGY_MANAGER
+    # TODO: permissioned: STRATEGY_MANAGER
 
-    # TODO: add strategy migrated event? 
-    assert self.strategies[old_strategy].activation != 0
-    assert self.strategies[old_strategy].currentDebt == 0
+    assert self.strategies[old_strategy].activation != 0, "old strategy not active"
+    assert self.strategies[old_strategy].currentDebt == 0, "old strategy has debt"
```

Comment:
this is not currently being tested: we want to eventually allow migrations with strategies that have debt right, and this is just a placeholder?

Reply:
you are right. this should be removed (I removed in other commit I think)